test/integration/framework/etcd.go

```diff
@@ -74,7 +74,8 @@ func startEtcd(output io.Writer) (func(), error) {
 	}
 	klog.V(1).Infof("could not connect to etcd: %v", err)
 
-	currentURL, stop, err := RunCustomEtcd("integration_test_etcd_data", nil, output)
+	etcdArgs := []string{"--experimental-watch-progress-notify-interval", "5s"}
```

Comment:
@p0lyn0mial if this needed to function properly, I'd like to see the code for the error case and a test that shows that the stream list request is properly rejected before we merge something that will make our test instances function.

Reply:
I believe this is a more generic issue. Currently, Kube doesn’t have a way to dynamically discover features of the underlying store. We rely on administrators to enable the required features.
For this specific feature, we added a fallback mechanism to the reflector, which will revert to a regular list in case of any issues, (e.g. a timeout). [Link to the code](https://github.com/kubernetes/kubernetes/pull/120984).

I believe it’s a bug that the integration tests didn’t set this flag, considering that `WatchBookmark` requires it and became `GA` in `1.17`.

I was also planning to highlight in the release notes and blog post that the flag is mandatory, but perhaps that isn’t sufficient?

In the future, it would be beneficial if we had some sort of automatic way to detect the features of the underlying store so that the system can adjust automatically.
For example for the `WatchList` we were planing to use https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/interfaces.go#L253

Reply:
> @p0lyn0mial if this needed to function properly, I'd like to see the code for the error case and a test that shows that the stream list request is properly rejected before we merge something that will make our test instances function.

This is still outstanding.  As I understand it, the issue isn't "detect what features are available".  The issue is does our client that is consuming the StreamingList today properly fail if the StreamingList is out of order?  Such failure is logically required if we don't order at the client and I seem to recall you saying this was necessary to be sure the entire list was present.  Is that distinct from the ordering?

Reply:
With manual progress notify [that we will shre between consistent lists and streaming lists] - bumping this flag to 5s is no longer needed.

Reply:
OK, let's close this PR then.